### DISCUSSION TITLE
Access auto-generated navigation category as document metadata?

### DISCUSSION THREAD

**Discussion by capnrefsmmat** [quarto-dev/quarto-cli#5181] [Feature Requests]
I'm organizing a site using the [auto-generated sidebar feature](https://quarto.org/docs/websites/website-navigation.html#auto-generation), so for instance I might have several files in directories like this:

```
biology/
  anage.qmd
  bug-attraction.qmd
money/
  ames-housing.qmd
  rail-trails.qmd
```

In my `_quarto.yml`, I have

```yaml
website:
  sidebar:
    contents: auto
    collapse-level: 1
```

So I get headings for Biology and Money and so on. I use this as sort of a category system. (Yes, I know Quarto also has categories, but I need two parallel category systems; I use the directories for the subject area of each page, and I use the `categories` field for the statistical methods involved in each page.)

I also have a [listing page](https://quarto.org/docs/websites/website-listings.html) listing all the documents in the website, so they can be sorted by various attributes. One can [choose listing fields to display](https://quarto.org/docs/websites/website-listings.html#listing-fields); so, for instance, if the document metadata has a `foo` attribute, I can have the listing table show a column for `foo`.

I'd like a column for the navigation category each document is in. That is, the row for `anage.qmd` should say it is in Biology, while `ames-housing.qmd` should say it is in Money.

Is it possible to get this information as automatically produced document metadata? Various listing fields are already automatically populated by Quarto, so it'd be nice to be able to get this one.

**Comment by mcanouil** [comment]
I have to say I have difficulties to fully understand what you want.
If you have a more concrete example and/or a sketch of the output it would help.

Maybe, for you it's a matter of generated yaml file for use in listing and using <https://quarto.org/docs/websites/website-listings.html#including-or-excluding-items> feature.

**Comment by capnrefsmmat** [comment]
Well, let me give a concrete example. I'm collecting datasets for classroom use. Each has a Qmd describing it, and there's a listing page showing them all: <https://cmustatistics.github.io/data-repository/by-method.html>

You can see on the left that the datasets are organized into groups by subject area. That's done by putting them in directories like `astronomy/` and `biology/`.

I'd like to add that subject area to the listing table, so e.g. the first row for AnAge says it's in Biology. But it's not part of the article metadata -- it's just which directory each file is in -- so I can't add it by configuring the listing fields.

**Comment by mcanouil** [comment]
Firstly, there isn't currently an integrated way of doing what you want.

Secondly, what's the deal with adding a field (a.k.a., metadata) in the article? The articles do not seem to be automatically generated. I think I am missing something ...
![image](https://user-images.githubusercontent.com/8896044/231895426-f4dca800-3178-4bf8-a2a5-f9995e5141bc.png)
The page/article template does not use the "domain" field for example, thus you don't see it in the article.
<img width="1510" alt="image" src="https://user-images.githubusercontent.com/8896044/231895572-82897a4c-ea1d-4c33-8fba-c0aac728df7d.png">

**Comment by capnrefsmmat** [comment]
Yes, I'm sure I could add a field for it. It just seems redundant for the domain to be automatically inferred by Quarto (for the left navigation), but the inferred domain isn't available anywhere, so it also has to be manually specified. It'd be nice for that automatically inferred domain to be available somewhere.

(I might also like to put it in the article template, so each article is marked with which domain it's in.)

Another option would be for the automatic navigation feature to support using a metadata field in each document instead of inferring from directories -- then I can specify the field and use it in listings or anywhere else I want.

**Reply by mcanouil** [reply]
The two elements are distinct: the sidebar is simply URIs to contents following directory tree, the listing follows a template and is not tied up to the sidebar.
Now you can create a custom listing that automatically fills the "domain" field with the parent directory of the Quarto documents, see <https://quarto.org/docs/websites/website-listings-custom.html>.
